brody_lab_to_nwb/msortedprocesseddatainterface.py

```diff
@@ -0,0 +1,143 @@
+        mat_data = dict(
+            start_times=mat_file["Msorted"]["Trials"]["stateTimes"]["sending_trialnum"][()][0],
+            stop_times=mat_file["Msorted"]["Trials"]["stateTimes"]["cleaned_up"][()][0],
+            trial_type=[chr(x) for x in mat_file["Msorted"]["Trials"]["trial_type"][0]],
+            violated=mat_file["Msorted"]["Trials"]["violated"][()][0].astype(bool),
+            is_hit=mat_file["Msorted"]["Trials"]["is_hit"][()][0].astype(bool),
+            side=[side_mapping[chr(x)] for x in mat_file["Msorted"]["Trials"]["sides"][0]],
+            gamma=mat_file["Msorted"]["Trials"]["gamma"][()][0],
+            reward_location=mat_file["Msorted"]["Trials"]["reward_loc"][()][0],
+            poked_r=mat_file["Msorted"]["Trials"]["pokedR"][()][0],
+            stim_dur_s=mat_file["Msorted"]["Trials"]["stim_dur_s"][()][0],
+            click_diff_hz=mat_file["Msorted"]["Trials"]["stim_dur_s"][()][0]
+        )
+
+        times_column_descriptions = dict(
+            wait_for_cpoke="",  # TODO
+            cpoke_in="",  # TODO
+            cpoke_out="",  # TODO
+            clicks_on="",  # TODO
+            clicks_off="",  # TODO
+            spoke="",  # TODO
+            right_reward="The time when right reward occured in seconds.",
+            left_reward="The time when left reward occured in seconds.",
+            error="The time when error occured in seconds."
+        )
+        times_column_descriptions.update({"break": ""})  # TODO
+        for col, description in times_column_descriptions.items():
+            name = f"{col}_times"
+            mat_data.update({name: mat_file["Msorted"]["Trials"]["stateTimes"][col][()][0]})
+            nwbfile.add_trial_column(name=name, description=description)
+
+        column_descriptions = dict(
+            trial_type="The identifier value for the trial type.",
+            violated="Binary identifier value for trial violation.",
+            is_hit="Binary identifier value for trial hits.",
+            side="Left or right.",
+            gamma="",  # TODO
+            reward_location="Location of the reward.",
+            poked_r="",  # TODO
+            stim_dur_s="Duration of stimuli in seconds.",
+            click_diff_hz=""  # TODO
+        )
+        for col, description in column_descriptions.items():
+            nwbfile.add_trial_column(name=col, description=description)
+
+        side_mapping = dict(l="left", r="right")
+        n_trials = len(mat_data["trial_type"])
+        add_pharma = mat_file["Msorted"]["Trials"]["pharma"]["manip"].shape == (1, n_trials)
+        add_laser = any(mat_file["Msorted"]["Trials"]["laser"]["isOn"][()][0])
+
+        if add_pharma:
+            mat_data.update(
+                pharma_manipulation=mat_file["Msorted"]["Trials"]["pharma"]["manip"],
+                pharma_injector_mm=mat_file["Msorted"]["Trials"]["pharma"]["injector_mm"][()][0],
+                pharma_dose_ng=mat_file["Msorted"]["Trials"]["pharma"]["doseNG"][()][0]
```

Comment:
Why are you fetching all the data and then taking the first element? Wouldn't `mat_file["Msorted"]["Trials"]["pharma"]["doseNG"][0]` make more sense?

Reply:
Fixed!